Calling/ClientApp/src/components/FeedbackButton.tsx

```diff
@@ -0,0 +1,203 @@
+  type: DialogType.largeHeader,
+  title: 'Feedback',
+  subText: 'Something went wrong? Please provide feedback here!'
+};
+
+const FeedbackPopup = (props: FeedbackPopupProps): JSX.Element => {
+  const [feedbackType, setFeedbackType] = useState<string>('');
+  const [comment, setComment] = useState<string>('');
+  const [guid, setGuid] = useState<string>('');
+  const [isScreenShotDialogOn, setIsScreenShotDialogOn] = useState<boolean>(false);
+  const [screenShot, setScreenShot] = useState<HTMLCanvasElement>();
+  const [isHidden, setIsHidden] = useState<boolean>(false);
+
+  const takeScreenShot = useCallback(async () => {
+    const screenShot = await captureScreenshot();
+    setScreenShot(screenShot);
+  }, []);
+
+  const submitFeedback = useCallback(async () => {
+    const feedback = createFeedback(feedbackType, comment);
+    await uploadFeedback(feedback, screenShot);
+    setGuid(feedback.guid);
+  }, [feedbackType, comment, screenShot]);
+
+  const feedbackInput = (
+    <>
+      <Label>What is this related to? (Required)</Label>
+      <Dropdown
+        defaultSelectedKey={'Calling - Audio'}
+        options={options}
+        onChange={(_, item): void => {
+          setFeedbackType(item?.key as string);
+        }}
+      />
+      <Label>What are you seeing? Has it always been that way? Any step to repro?</Label>
+      <TextField
+        onChange={(_, value): void => {
+          setComment(value ?? '');
+        }}
+        multiline
+        rows={6}
+      />
+      <Checkbox
+        label="Include screen shot"
+        checked={!!screenShot}
+        onChange={(_, value): void => (value ? setIsScreenShotDialogOn(value ?? false) : setScreenShot(undefined))}
+      />
+      {isScreenShotDialogOn && (
+        <ScreenShotInstruction
+          onDismiss={(): void => {
+            setIsScreenShotDialogOn(false);
+          }}
+          onTakeScreenShot={takeScreenShot}
+          setIsParentHidden={setIsHidden}
+        />
+      )}
+      {screenShot && <img src={screenShot?.toDataURL()} style={screenShotStyle} alt='screen shot'/>}
+    </>
+  );
+
+  const feedbackSubmitted = (
+    <>
+      <Label>Please provide this guid to developers for targeting the case:</Label>
+      <TextField value={guid} readOnly />
+    </>
+  );
+
+  return (
+    <Dialog
+      dialogContentProps={dialogContentProps}
+      modalProps={modalProps}
+      hidden={isHidden}
+      onDismiss={props.toggleHidden}
+    >
+      {!guid && feedbackInput}
+      {guid && feedbackSubmitted}
+      <DialogFooter>
+        {!guid && <PrimaryButton onClick={submitFeedback} text="Send" />}
+        <DefaultButton onClick={props.toggleHidden} text="Close" />
+      </DialogFooter>
+    </Dialog>
+  );
+};
+
+const createFeedback = (feedbackType: string, comment: string): Feedback => {
+  const callId = store.getState().calls.call?.id ?? 'Not Found';
+  return {
+    logs: getLogs(),
+    callId: callId,
+    comments: comment,
+    guid: createGUID(),
```

Comment:
optional: guid is just a type of string, "correlationId" might make more sense.

Reply:
I remember call sdk have a data row named correlationId, maybe I can rename it as feedbackId to avoid duplication